### PR TITLE
KTOR-6447 Update Kotlin to 1.9.20

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-6447](https://youtrack.jetbrains.com/issue/KTOR-6447/Update-Kotlin-to-1.9.20)

### CODE REVIEW

**Review by rsinukov — CHANGES_REQUESTED** [review]

File: build.gradle.kts [hunk 1]
```diff
@@ -151,7 +155,7 @@ fun configureDokka() {
 
         val dokkaPlugin by configurations
         dependencies {
-            dokkaPlugin("org.jetbrains.dokka:versioning-plugin:1.9.0")
+            dokkaPlugin("org.jetbrains.dokka:versioning-plugin:1.9.10")
```
[the remark this answers is not in the document]
Comment (e5l) [reply build.gradle.kts:158]: dokka 1.9.20 is not released yet

File: gradle.properties [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 # ktor
 # Makes Ktor import and resolve
-ktor.ide.jvmAndCommonOnly=true
+ktor.ide.jvmAndCommonOnly=false
```
[the remark this answers is not in the document]
Comment (e5l) [reply gradle.properties:7]: nope, will revert

File: gradle.properties [hunk 3]
```diff
@@ -17,6 +17,7 @@ org.gradle.daemon=true
 org.gradle.caching=true
 org.gradle.vfs.watch=true
 org.gradle.jvmargs=-Xmx6g -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8 -XX:+UseParallelGC
+# org.gradle.configuration-cache=true
```
[the remark this answers is not in the document]
Comment (e5l) [reply gradle.properties:20]: thanks, will drop, it's not compatible anyway

File: ktor-network/nix/src/io/ktor/network/sockets/SocketAddressNative.kt [hunk 4]
```diff
@@ -2,11 +2,50 @@ package io.ktor.network.sockets
 
 public actual sealed class SocketAddress
 
-public actual data class InetSocketAddress actual constructor(
+public actual class InetSocketAddress actual constructor(
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-network/nix/src/io/ktor/network/sockets/SocketAddressNative.kt:5]: Thanks, will add

**Review by rsinukov — APPROVED** [review]

**Comment by IgnatBeresnev** [comment]
Since Dokka was updated to 1.9.10, JFYI this workaround is no longer needed and can be removed

https://github.com/ktorio/ktor/blob/b4a2f0730d0de4e95a7c9f6beac16f7b8dc479f4/gradle.properties#L37-L39

cc @rsinukov @e5l

**Comment by e5l** [comment]
Thanks, will update.

**Pull request merged**